File: services/go/api/v1/users/getUser.go
package users

import (
	"api/utils"
	. "api/v1/commons"
	"errors"

	"github.com/gofiber/fiber/v2"
	"gorm.io/gorm"
)

func FindUser(c *fiber.Ctx) error {
	var username = c.Params("username")

	user, err := utils.GetUserPartial(username)

	if err != nil {
		if errors.Is(err, gorm.ErrRecordNotFound) {
			return c.Status(404).JSON(Response{
				Success: false,
				Message: "User not found.",
			})
		}

		return c.Status(500).JSON(Response{
			Success: false,
			Message: err.Error(),
		})
	}

	return c.JSON(Response{
		Success: true,
		Data:    user,
	})
}

File: services/node/api/src/handlers/users/getUser.ts
import { eq } from "drizzle-orm/expressions";
import { db } from "../../db";
import { users } from "../../db/schemas";
import { FastifyImp, User } from "../../types";
import { USER_PUBLIC_OBJECT } from "../../utils/publicObjects";

export const getUser: FastifyImp<
  {},
  unknown,
  unknown,
  {
    username: string;
  }
> = async (request, reply) => {
  const username = request.params.username;

  const user: User =
    (
      await db
        .select(USER_PUBLIC_OBJECT)
        .from(users)
        .where(eq(users.username, username))
    )[0] ?? null;

  if (!user) {
    reply.status(404).send({
      success: false,
      error: {
        message: "User not found",
      },
    });
  }

  reply.send({
    success: true,
    data: user,
  });
};
